Add watch option to 'ank get workloads' command (#463)
eclipse-ankaios/ankaios: ank/doc/swdesign/README.md

- impl
- utest

#### CLI shall provides continuous workload monitoring with watch mode
`swdd~cli-get-workloads-with-watch~1`

When the CLI requests a list of workload with the `watch` option enabled, the CLI shall continuously stream real-time workload updates until manually terminated by the user by following this steps:
- Fetch and display the initial list of workloads from the Ankaios Server, applying user-defined filters (agent, state, workload names).
- Continuously listen for `UpdateWorkloadState` messages and apply workload updates.
- Dynamically update the table by:
    - Adding newly created workloads.
    - Removing workloads marked as `Removed`.
    - Modifying existing workloads' execution states or metadata.
- Optimize rendering to refresh only affected table rows, preventing full-screen reload.

> - Execute infinitely until explicitly terminated by the user using `Ctrl+C`.

Reviewer: Thank you making the swdd changes.

I have a few further suggestions:
- the first action might be reduced a little bit to something like `..., the CLI shall continuously watch for workload changes by:`
- The part while listening to workload state updates (bullet point 2 and 3) are related to each other and the dynamic workload table updates underly some conditions. I suggest to split the swdd into two swdds to add the conditions properly. The second one separates the "process_workload_updates" logic while the first one says that it continuously listens to workload state updates to update the workload table data.
- the bullet point with `Optimize rendering to refresh only ...` can be removed
- the last bullet point with `Ctrl+C` we can put as comment
- the filtering logic can be also mentioned in a comment

Would you do the changes again? If you need support or detailed suggestions let me know.

Author: Thanks for your comments! I appreciate the feedback and always happy to improve to achieve best practices.

Reviewer: The swdd requirements have been adapted in [10feffa](https://github.com/eclipse-ankaios/ankaios/pull/463/commits/10feffa6a04c224d115474f2b6b8aa41ca5e1f71) and further commits.

- impl
- utest

#### CLI shall provides continuous workload monitoring with watch mode
`swdd~cli-get-workloads-with-watch~1`

When the CLI requests a list of workload with the `watch` option enabled, the CLI shall continuously watch for workload changes by:
- Fetch and display the initial list of workloads from the Ankaios Server.
- Continuously listen for `UpdateWorkloadState` messages.
- Dynamically update the table by:
    - Adding newly created workloads.
    - Removing workloads marked as `Removed`.
    - Modifying existing workloads' execution states or metadata.

> Comment:
> - User termination via `Ctrl+C`.
> - Supports user-defined filters (agent, state, names).

Reviewer: ```suggestion
Comment:
The `watch` mode executes infinitely until the user explicitly terminates it using `Ctrl+C`.
All received workload updates are filtered with user-defined parameters (agent, state, workload names).
```

Reviewer: Taken over in [34a0779](https://github.com/eclipse-ankaios/ankaios/pull/463/commits/34a077965feba681c4f45726d6f64e6cfe71a947)

- impl
- utest

#### CLI initiates continuous workload monitoring
`swdd~cli-get-workloads-with-watch-init~1`

When the CLI requests a list of workloads with the `watch` option enabled, the CLI shall continuously watch for workload changes by:

- requesting the initial list of workloads from the Ankaios Server
- displaying the workloads in an alphabetically sorted table
- updating the initial list of workloads by continuously listening for `UpdateWorkloadState` messages

> Comment: The `watch` mode executes infinitely until the user explicitly terminates it using `Ctrl+C`. All received workload data is filtered with user-defined parameters (agent, state, workload names).

Reviewer: I commit this directly. After `comment` we need a newline.

- impl
- utest

#### CLI initiates continuous workload monitoring
`swdd~cli-get-workloads-with-watch-init~1`

When the CLI executes a workloads request with watch option enabled, the CLI shall:

- Fetch initial workload states from Ankaios Server
- Display the workloads in a sorted alphabetically table format.
- Establish persistent connection to UpdateWorkloadState stream
- Process updates continuously until user termination

Comment: The `watch` mode executes infinitely until the user explicitly terminates it using `Ctrl+C`. All received workload data is filtered with user-defined parameters (agent, state, workload names).

Tags:
- Cli
- CliCommands

Needs:
- impl
- utest

#### CLI processes workload state updates for watch mode
`swdd~cli-process-workload-updates~1`

> When processing UpdateWorkloadState messages during watch mode, the CLI shall:

Reviewer: ```suggestion
When processing `UpdateWorkloadState` messages during watch mode, for each workload state the CLI shall:
```

> - Add new workloads: Full state re-fetch on unknown instances.
> - Update existing workloads: Modify workload data then revalidate filters.
> - Remove workloads: Immediate deletion on "Removed" state detection.
> - Reapply filters after every update cycle.

Reviewer: ```suggestion
* either update the current list of workloads with a new list of workloads from the Ankaios Server if the received workload state is from a newly added workload
* or delete workload from the current list if the received state is `Removed`
* or update the current state with new state inside the current list
```

> Comment: Filters for unknown instances are applied prior to transitioning to the BTreeMap to ensure that workloads not matching the filters are excluded from insertion.

Reviewer: ```suggestion
Comment:
All received workload data is filtered with user-defined parameters (agent, state, workload names).
```
